packages/web/src/pages/settings-page/components/mobile/ChangePasswordPage.tsx

```diff
@@ -1,61 +1,125 @@
+  close: 'Close'
+}
 
-  const currentPage = useSelector(getCurrentPage)
+const ChangePasswordMobileForm = ({
+  page,
+  onClose
+}: {
+  page: ChangePasswordPage
+  onClose: () => void
+}) => {
+  const { isSubmitting, isValid } = useFormikContext<ChangePasswordFormValues>()
+  return (
+    <Flex
+      as={Form}
+      direction={'column'}
+      justifyContent='space-between'
+      css={{ flexGrow: 1 }}
+    >
+      <Flex p='l' gap={'s'} direction='column' mt={'xl'}>
+        <Text variant='heading' color='heading'>
+          {messages.changeYourPassword}
+        </Text>
+        <SlidingPages currentPage={page}>
+          <ConfirmCredentialsPage />
+          <VerifyEmailPage />
+          <NewPasswordPage />
+          <SuccessPage />
+        </SlidingPages>
+      </Flex>
+      <Paper shadow={'midInverted'} p={'l'}>
+        {page === ChangePasswordPage.Success ? (
+          <Button fullWidth variant='primary' onClick={onClose} type='button'>
+            {messages.close}
+          </Button>
+        ) : page === ChangePasswordPage.NewPassword ? (
+          <Button
+            fullWidth
+            variant='primary'
+            iconRight={IconLock}
+            type='submit'
+            isLoading={isSubmitting}
+            disabled={!isValid}
+          >
+            {messages.changePassword}
+          </Button>
+        ) : (
+          <Button
+            fullWidth
+            variant='primary'
+            iconRight={IconArrowRight}
+            type={'submit'}
+            isLoading={isSubmitting}
+          >
+            {messages.continue}
+          </Button>
+        )}
+      </Paper>
+    </Flex>
+  )
+}
+
+export const ChangePasswordMobilePage = ({ goBack }: SettingsPageProps) => {
+  const navContext = useContext(NavContext)!
 
-  // Remove back arrow on new password and loading pages
   useEffect(() => {
-    if (
-      [
-        ChangePasswordPageStep.NEW_PASSWORD,
-        ChangePasswordPageStep.LOADING
-      ].includes(currentPage)
-    ) {
-      navContext.setLeft(null)
-    } else {
+    navContext.setLeft(LeftPreset.CLOSE)
+    return () => {
       navContext.setLeft(LeftPreset.BACK)
     }
-  }, [navContext, currentPage])
+  }, [navContext])
 
   // Go back to account settings when done
   const onComplete = useCallback(() => {
     goBack()
   }, [goBack])
 
-  // On initial render, set the page to confirm credentials
-  useEffect(() => {
-    dispatch(changePage(ChangePasswordPageStep.CONFIRM_CREDENTIALS))
-    const trackEvent: TrackEvent = make(Name.SETTINGS_START_CHANGE_PASSWORD, {})
-    dispatch(trackEvent)
-  }, [dispatch])
+  const { page, ...formikConfiguration } = useChangePasswordFormConfiguration()
 
   return (
-    <div className={styles.container}>
-      <ChangePassword isMobile={true} onComplete={onComplete} />
-    </div>
+    <Flex
+      direction='column'
+      css={{
+        textAlign: 'start',
+        backgroundColor: 'white',
+        height: 'calc(100% - 50px)'
```

Comment:
why this? though i realize not new code...

Reply:
I'll add a comment - it accounts for the header